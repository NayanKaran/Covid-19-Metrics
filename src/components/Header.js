import React from 'react';
import { useLocation, useNavigate } from 'react-router-dom';

const Header = () => {
  const location = useLocation();
  const navigate = useNavigate();

  const goBack = () => navigate(-1);

  const handleKeyEvent = (event) => {
    const { key } = event;
    if (key === 'Backspace') {
      navigate(-1);
    }
  };

  return (
    <header style={{
      backgroundColor: '#eb4c89', display: 'flex', alignItems: 'center', height: '56px', color: 'white',
    }}
    >
      <div
        style={{
          padding: '0 8px',
        }}
        className={location.pathname === '/' ? 'hidden' : undefined}
        onClick={goBack}
        onKeyUp={handleKeyEvent}
        role="button"
        tabIndex="0"
      >
        <i className="fa-solid fa-angle-left" style={{ color: 'white' }} />
      </div>
      <span style={{
        margin: '0 auto',
        paddingBottom: '4px',
        fontSize: '24px',
      }}
      >
        {location.pathname === '/' ? 'global cases' : 'continent cases'}
      </span>
    </header>
  );
};

export default Header;
