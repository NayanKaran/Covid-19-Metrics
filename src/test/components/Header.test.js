import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import Header from '../../components/Header';

describe('Testing Header component', () => {
  it('Testing with snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <Header />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
