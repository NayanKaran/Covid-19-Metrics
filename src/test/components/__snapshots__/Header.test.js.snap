// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header component Testing with snapshot 1`] = `
<header
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#eb4c89",
      "color": "white",
      "display": "flex",
      "height": "56px",
    }
  }
>
  <div
    className="hidden"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="button"
    style={
      Object {
        "padding": "0 8px",
      }
    }
    tabIndex="0"
  >
    <i
      className="fa-solid fa-angle-left"
      style={
        Object {
          "color": "white",
        }
      }
    />
  </div>
  <span
    style={
      Object {
        "fontSize": "24px",
        "margin": "0 auto",
        "paddingBottom": "4px",
      }
    }
  >
    global cases
  </span>
</header>
`;
